frontend/src/views/PrinterControlPage.vue

```diff
@@ -112,25 +112,32 @@
           </div>
         </div>
         <div class="stream-container">
-          <div v-if="webcams.length > 1" class="webcam-selector">
-            <select id="webcam-select" v-model="selectedWebcamIndex" class="custom-select">
-              <option v-for="(webcam, index) in webcams" :key="index" :value="index">
-                {{ webcam.name || '_default_' }}
-              </option>
-            </select>
+          <div class="header-container">
+            <div class="title">{{ $t('Webcam') }}</div>
+            <div class="webcam-options">
+              <b-dropdown v-model="selectedWebcamIndex" class="webcam-dropdown" block size="sm" variant="link" toggle-class="text-decoration-none" no-caret>
+                <template #button-content>
+                  <i class="fa fa-camera" aria-hidden="true"></i>
+                  {{ isAllWebcamSelected ? 'All' : selectedWebcam.name || 'Primary' }}
+                  <i class="fa fa-chevron-down" aria-hidden="true"></i>
+                </template>
+                <b-dropdown-text class="small text-secondary">{{ $t("WEBCAM SELECTION") }}</b-dropdown-text>
+                <b-dropdown-item v-for="(webcam, index) in webcams" :key="index" href="#" @click="chooseWebcam(index)">{{ webcam.name || 'Primary' }}</b-dropdown-item>
+                <b-dropdown-item @click="chooseWebcam('all')">{{ $t('All') }}</b-dropdown-item>
+              </b-dropdown>
+            </div>
+          </div>
+          <div class="webcam-main" :class="{ 'justify-center' : isAllWebcamSelected, 'webcam-more-than-two' : isAllWebcamSelected && webcams.length > 2 }">
+            <div v-for="(webcam, index) in webcams" :key="index" ref="streamInner" class="stream-inner" :class="isAllWebcamSelected ? (isAtleastOnePrinterPortrait ? 'two-webcam-portrait' : 'two-webcam-landscape') : (videoRotationDeg === 90 || videoRotationDeg === 270 ? 'single-webcam-portrait' : '')" v-show="isAllWebcamSelected ? true : (index === selectedWebcamIndex)">
+              <streaming-box
+                :printer="printer"
+                :webrtc="webcam.webrtc"
```

Comment:
If we are passing webcam to stream-box, then there is no need to pass webcam.webrtc to it again, right?

Reply:
streaming-box component is being on other places as well, they depend upon webrtc and stream-box is written to take webrtc from before.